OpenRA.Mods.RA2/Traits/AttackInfect.cs

```diff
@@ -0,0 +1,105 @@
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.RA2.Activities;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("Move onto the target then execute the attack.")]
+	public class AttackInfectInfo : AttackFrontalInfo, Requires<MobileInfo>
+	{
+		[Desc("Range of the final joust of the infector.")]
+		public readonly WDist JoustRange = WDist.Zero;
+
+		[Desc("Conditions that last from start of the joust until the attack.")]
+		[GrantedConditionReference]
+		public readonly string JoustCondition = "jousting";
+
+		[FieldLoader.Require]
+		[Desc("How much damage to deal.")]
+		public readonly int Damage = 0;
+
+		[FieldLoader.Require]
+		[Desc("How often to deal the damage.")]
+		public readonly int DamageInterval = 0;
+
+		[Desc("Damage types for the infection damage.")]
+		public readonly BitSet<DamageType> DamageTypes = default(BitSet<DamageType>);
+
+		[Desc("If an external actor delivers more damage than this value, the infector is killed immediately.",
+			"Use -1 to never kill the infector.")]
+		public readonly int SuppressionDamageThreshold = -1;
+
+		[Desc("If the infected actor receives more damage from external sources than this value, the infector is killed immediately.",
+			"Use -1 to never kill the infector.")]
+		public readonly int SuppressionSumThreshold = -1;
+
+		[Desc("If the infected actor receives damage more times from external sources than this value, the infector is killed immediately.",
+			"Only counted if the value is above 0.")]
+		public readonly int SuppressionCountThreshold = 0;
+
+		[Desc("Damage type used for the suppression calculations.")]
+		public readonly BitSet<DamageType> SuppressionDamageType = default(BitSet<DamageType>);
+
+		[Desc("Damage types which allows the infector survive when it's host dies.")]
+		public readonly BitSet<DamageType> SurviveHostDamageTypes = default(BitSet<DamageType>);
+
+		public override object Create(ActorInitializer init) { return new AttackInfect(init.Self, this); }
+	}
+
+	public class AttackInfect : AttackFrontal
+	{
+		ConditionManager conditionManager;
+		int joustToken = ConditionManager.InvalidConditionToken;
+
+		public AttackInfect(Actor self, AttackInfectInfo info)
+			: base(self, info) { }
+
+		protected override void Created(Actor self)
+		{
+			conditionManager = self.TraitOrDefault<ConditionManager>();
+			base.Created(self);
+		}
+
+		protected override bool CanAttack(Actor self, Target target)
+		{
+			if (target.Type != TargetType.Actor)
+				return false;
+
+			if (self.Location == target.Actor.Location && HasAnyValidWeapons(target))
+				return true;
+
+			return base.CanAttack(self, target);
+		}
+
+		public void GrantJoustCondition(Actor self)
+		{
+			var info = Info as AttackInfectInfo;
```

Comment:
It would likely make sense to expose this as public property, considering we have this cast pattern in multiple places.

Reply:
Good idea. Done. This is a weird pattern when `Info` is already something else.